### PR TITLE
feat(docs): add useForm documentation

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
http://refine.dev/docs/api-reference/core/hooks/useForm

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

**Comment by salihozdemir** [comment]
When I'm reviewing your doc, I saw that:

"If you're looking for a complete form library, Refine supports two form libraries out-of-the-box."

It must be "..., **refine** supports three form ..."

I've created comment because your changes not cover it 😅

File: documentation/docs/api-reference/core/hooks/useForm.md [hunk 1]
```diff
@@ -45,37 +672,740 @@ export const PostCreate = () => {
...
+```
+
+### `invalidates`
+
+You can use it to manage the invalidations that will occur at the end of the mutation.
+
+By default it's invalidates following queries from the current `resource`:
+
+-   on `create` or `clone` mode: `"list"` and `"many"`
+-   on `edit` mode: `"list`", `"many"` and `"detail"`
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/useForm.md:1208]:
```suggestion
-   on `create` or `clone` mode: `"list"`, and `"many"`
-   on `edit` mode: `"list"`, `"many"`, and `"detail"`
```

File: documentation/docs/api-reference/core/hooks/useForm.md [hunk 2]
```diff
@@ -45,37 +672,740 @@ export const PostCreate = () => {
...
+
+### `dataProviderName`
+
+If there is more than one dataProvider, you should use the dataProviderName that you will use.
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/useForm.md:1212]:
```suggestion
If there is more than one dataProvider, you should use the `dataProviderName` that you will use.
```

File: documentation/docs/api-reference/core/hooks/useForm.md [hunk 3]
```diff
@@ -45,37 +672,740 @@ export const PostCreate = () => {
...
+]}>
+<TabItem value="create">
+
+`action: "create"` is used for creating a new record that didn't exist before.
+
+`useForm` uses [`useCreate`](/docs/api-reference/core/hooks/data/useCreate.md) under the hood for mutations on create mode.
+
+In the following example, we'll show how to use `useForm` with `action: "create"`.
+
+```tsx live tailwind url=http://localhost:3000 previewHeight=420px
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/useForm.md:706]:
```suggestion
```tsx live tailwind url=http://localhost:3000/posts/create previewHeight=420px
```

File: documentation/docs/api-reference/core/hooks/useForm.md [hunk 4]
```diff
@@ -45,37 +672,740 @@ export const PostCreate = () => {
...
+<TabItem value="edit">
 
 `action: "edit"` is used for editing an existing record. It requires the `id` for determining the record to edit. By default, it uses the `id` from the route. It can be changed with the `setId` function or `id` property.
 
-It fetches the record data according to the `id` and returns the `queryResult` for you to fill the form.
+It fetches the record data according to the `id` with [`useOne`](/docs/api-reference/core/hooks/data/useOne/) and returns the `queryResult` for you to fill the form. After the form is submitted, it updates the record with [`useUpdate`](/api-reference/core/hooks/data/useUpdate.md).
+
+In the following example, we'll show how to use `useForm` with `action: "edit"`.
+
+```tsx live tailwind url=http://localhost:3000 previewHeight=420px
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/useForm.md:812]:
```suggestion
```tsx live tailwind url=http://localhost:3000/posts/edit/123 previewHeight=420px
```

File: documentation/docs/api-reference/core/hooks/useForm.md [hunk 5]
```diff
@@ -45,37 +672,740 @@ export const PostCreate = () => {
...
+You can think `action:clone` like save as. It's similar to `action:edit` but it creates a new record instead of updating the existing one.
 
-`useForm` uses [`useUpdate`](/api-reference/core/hooks/data/useUpdate.md) under the hood for mutations on clone mode.
+It fetches the record data according to the `id` with [`useOne`](/docs/api-reference/core/hooks/data/useOne/) and returns the `queryResult` for you to fill the form. After the form is submitted, it creates a new record with [`useCreate`](/docs/api-reference/core/hooks/data/useCreate.md).
+
+In the following example, we'll show how to use `useForm` with `action: "clone"`.
+
+You can go to edit page by clicking on the edit button in the list page. You will see `action:clone` toggle at the top of the page. You can toggle it to set the action to `clone`.
+
+```tsx live tailwind url=http://localhost:3000 previewHeight=420px
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/hooks/useForm.md:931]:
```suggestion
```tsx live tailwind url=http://localhost:3000/posts/edit/123 previewHeight=420px
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**